src/dict.c

```diff
@@ -1474,10 +1474,10 @@ static void _dictShrinkIfNeeded(dict *d)
     /* If we reached below 1:10 elements/buckets ratio, and we are allowed to resize
      * the hash table (global setting) or we should avoid it but the ratio is below 1:50,
      * we'll trigger a resize of the hash table. */
-    if ((dict_can_resize == DICT_RESIZE_ENABLE &&  
-         d->ht_used[0] * 100 / DICTHT_SIZE(d->ht_size_exp[0]) < HASHTABLE_MIN_FILL) ||
+    if ((dict_can_resize == DICT_RESIZE_ENABLE &&
+         d->ht_used[0] * 100 <= HASHTABLE_MIN_FILL * DICTHT_SIZE(d->ht_size_exp[0])) ||
         (dict_can_resize != DICT_RESIZE_FORBID &&
-         d->ht_used[0] * 100 / DICTHT_SIZE(d->ht_size_exp[0]) < HASHTABLE_MIN_FILL / dict_force_resize_ratio))
+         d->ht_used[0] * 100 <= HASHTABLE_MIN_FILL / dict_force_resize_ratio * DICTHT_SIZE(d->ht_size_exp[0])))
```

Comment:
Do we want to avoid this integer division too?

Reply:
change `dict_force_resize_ratio` to 4?

Reply:
or move it to multiplication on the left side

```suggestion
         d->ht_used[0] * 100 * dict_force_resize_ratio <= HASHTABLE_MIN_FILL * DICTHT_SIZE(d->ht_size_exp[0])))
```